### PR TITLE
Refactor PrometheusExporter not hold internal HttpListener

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Works towards resolving #65. This commit breaks the internal dependency of HttpListener from the Exporter. Subsequent PRs will add alternative methods for exposing the metrics endpoint.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs [hunk 1]
```diff
@@ -13,30 +13,83 @@
...
+using OpenTelemetry.Exporter.Prometheus.Implementation;
 
-namespace OpenTelemetry.Exporter.Prometheus.Implementation
+namespace OpenTelemetry.Exporter.Prometheus
 {
-    internal class MetricsHttpServer
+    /// <summary>
+    /// A HTTP listener used to expose Prometheus metrics.
+    /// </summary>
+    public class PrometheusExporterMetricsHttpServer
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs]: make it disposable to indicate the need to `Stop` it

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs [hunk 2]
```diff
@@ -13,30 +13,83 @@
...
+        private Task workerThread;
 
-        public MetricsHttpServer(List<Metric> metrics, PrometheusExporterOptions options, CancellationToken token)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrometheusExporterMetricsHttpServer"/> class.
+        /// </summary>
+        /// <param name="exporter">The <see cref="PrometheusExporter"/> instance.</param>
+        /// <param name="options">The <see cref="PrometheusExporterOptions"/> instance.</param>
+        /// <param name="token">A <see cref="CancellationToken"/> that can be used to stop the Htto Server.</param>
+        public PrometheusExporterMetricsHttpServer(PrometheusExporter exporter, PrometheusExporterOptions options, CancellationToken token)
```
Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs]: `PrometheusExporter` was already created with the set of `PrometheusExporterOptions`. It's better to get config values from the exporter than pass them again here. This way the situation when one passed different set of properties to exporter and server will be eliminated

Comment (SergeyKanzhelev) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMetricsHttpServer.cs]: should the cancellation token be passed to `Start` than?

**Comment by MikeGoldsmith** [comment]
Updated HttpServer based on @SergeyKanzhelev's suggestions.

**Pull request merged**